### DISCUSSION TITLE
Difference between sync_to_async vs. database_sync_to_async

### DISCUSSION THREAD

**Discussion by keithhackbarth** [django/channels#2073] [General] [answered]
I understand that _database_sync_to_async_ explicitly closes all DB connections. But when I run _sync_to_async_ with Django 5.0 + Postgres - I don't see a difference in the total number of connections. They seem to have the same effect.

Curious what guidance here is? Should I just default to using _database_sync_to_async_ for all cases?

**Comment by carltongibson** [comment, marked as answer]
Hi @keithhackbarth, 

`database_sync_to_async()` literally just wraps `sync_to_async()` with a call to Django's (internal) clean up function `close_old_connections()`. 

Django itself calls that for each request when dispatching the [request_finised signal](https://docs.djangoproject.com/en/5.0/ref/signals/#django.core.signals.request_finished). 

So, if you're running your requests through to Django, likely you'll not need to call it yourself. For requests which go only to Channels consumers, and make use of the database, you likely should use `database_sync_to_async()`. 

Make sense?

**Reply by keithhackbarth** [reply]
@carltongibson - You are my hero! That's exactly what I was looking for. Thank you 👏 👏

**Reply by carltongibson** [reply]
@keithhackbarth I can’t ask for more than that. 🥰
